cmd/ethrex/l2/command.rs

```diff
@@ -269,142 +284,156 @@ impl Command {
+                    store_type,
+                    genesis.to_str().expect("Invalid genesis path"),
+                )
+                .await?;
+
+                let rollup_store = StoreRollup::new(
+                    store_path
+                        .join("./rollup_store")
+                        .to_str()
+                        .expect("Invalid store path"),
+                    rollup_store_type,
+                )?;
+                rollup_store
+                    .init()
+                    .await
+                    .expect("Failed to init rollup store");
+
+                // Get genesis
+                let genesis_header = store.get_block_header(0)?.expect("Genesis block not found");
+                let genesis_block_hash = genesis_header.hash();
+
+                let mut new_trie = store
+                    .state_trie(genesis_block_hash)?
+                    .expect("Cannot open state trie");
+
+                let mut last_block_number = 0;
+
+                // Iterate over each blob
+                let files: Vec<std::fs::DirEntry> = read_dir(blobs_dir)?.try_collect()?;
+                for (file_number, file) in files
+                    .into_iter()
+                    .sorted_by_key(|f| f.file_name())
+                    .enumerate()
+                {
+                    let batch_number = file_number as u64 + 1;
+                    let blob = std::fs::read(file.path())?;
+
+                    if blob.len() != BYTES_PER_BLOB {
+                        panic!("Invalid blob size");
+                    }
+
+                    // Decode state diff from blob
+                    let blob = bytes_from_blob(blob.into());
+                    let state_diff = StateDiff::decode(&blob)?;
+
+                    // Apply all account updates to trie
+                    let account_updates = state_diff.to_account_updates(&new_trie)?;
+                    let account_updates_list = store
+                        .apply_account_updates_from_trie_batch(new_trie, account_updates.values())
+                        .await
+                        .expect("Error applying account updates");
+
+                    let (new_state_root, state_updates, accounts_updates) = (
+                        account_updates_list.state_trie_hash,
+                        account_updates_list.state_updates,
+                        account_updates_list.storage_updates,
+                    );
+
+                    let pseudo_update_batch = UpdateBatch {
+                        account_updates: state_updates,
+                        storage_updates: accounts_updates,
+                        blocks: vec![],
+                        receipts: vec![],
+                        code_updates: vec![],
+                    };
+
+                    store
+                        .store_block_updates(pseudo_update_batch)
+                        .await
+                        .expect("Error storing trie updates");
```

Comment:
same here, it consumes the error reason

Reply:
[eb4cb38](https://github.com/lambdaclass/ethrex/pull/3826/commits/eb4cb3864cbe789bac984481a83664e7f28ea0a3)